app/src/main/java/com/github/sdp/ratemyepfl/viewmodel/AddReviewViewModel.kt

```diff
@@ -17,13 +23,24 @@ import javax.inject.Inject
  */
 @HiltViewModel
 class AddReviewViewModel @Inject constructor(
-    private val database: ReviewsRepositoryInterface
+    private val database: ReviewsRepositoryInterface,
+    private val itemRepo: ItemsRepositoryInterface,
+    private val savedStateHandle: SavedStateHandle
 ) : ViewModel() {
 
     val rating: MutableLiveData<ReviewRating> = MutableLiveData(null)
     val title: MutableLiveData<String> = MutableLiveData(null)
     val comment: MutableLiveData<String> = MutableLiveData(null)
     private var date: LocalDate? = null
+    private var item: Reviewable? = null
```

Comment:
I think that you can transform it into val. You retrieve the id before and then assign the value instead of having a var.

Reply:
I can't re-assign a value to a val.

Reply:
Never mind, it's a suspend function you can't.